src/ZkFold/Base/Algebra/Polynomials/Univariate.hs

```diff
@@ -59,9 +62,20 @@ newtype Poly c = P (V.Vector c)
 toPoly :: (Ring c, Eq c) => V.Vector c -> Poly c
 toPoly = removeZeros . P
 
+constant :: (Ring c, Eq c) => c -> Poly c
+constant = toPoly . V.singleton
+
+-- | A polynomial of form cx^d
+--
+monomial :: (Ring c, Eq c) => Natural -> c -> Poly c
+monomial d c = toPoly $ V.fromList (replicate d zero P.<> [c])
+
 fromPoly :: Poly c -> V.Vector c
 fromPoly (P cs) = cs
 
+evalPoly :: Ring c => Poly c -> c -> c
+evalPoly (P cs) x = sum $ V.zipWith (*) cs $ fmap (x^) (V.generate (V.length cs) (fromIntegral @_ @Natural))
```

Comment:
Urges to introduce some common interface to different kinds of polynomials, will add a proposal about this later

Reply:
Written down in https://github.com/zkFold/zkfold-base/issues/154#issuecomment-2337702244